internal/pkg/mapper/mapper.go

```diff
@@ -0,0 +1,255 @@
+package mapper
+
+import (
+	"fmt"
+	"regexp"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	aggregationv1 "github.com/envoyproxy/xds-relay/pkg/api/aggregation/v1"
+)
+
+type rule = aggregationv1.KeyerConfiguration_Fragment_Rule
+type matchPredicate = aggregationv1.MatchPredicate
+type repeatedResultPredicate = aggregationv1.ResultPredicate_RepeatedResultPredicate
+
+// Mapper defines the interface that Maps an incoming request to an aggregation key
+type Mapper interface {
+	GetKey(node core.Node, typeURL string) (string, error)
+}
+
+type mapper struct {
+	config aggregationv1.KeyerConfiguration
```

Comment:
would use the logger defined in `/internal/pkg/log` and store it here, replacing the `fmt` calls.

(I'm separately considering placing an interface in front of the zap logger but that's a different conversation)

Reply:
`fmt` is used to create `Error` objects. Will look at zap to see if it supports a way to create error objects.